engine.py

```diff
@@ -58,7 +58,16 @@ def get_pawn_moves(self, r, c, moves):
 			
 			for i in range(len(self.board)):
 				for j in range(len(self.board[i])):
```

Comment:
nice code! However the code might need a little refactoring to make it more efficient. As it is now, it checks every single board position from top to bottom to see if it is rank 2 or 3 for pawn moves which really isn't necessary.